scipy/optimize/tests/test__optimize.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+import pytest
+from numpy.testing import assert_allclose, assert_
+
+from scipy.optimize.optimize import MemoizeJac
+
+class FunctionWithGradient(object):
+    def __init__(self):
+        self.number_of_calls = 0
+
+    def __call__(self, x):
+        self.number_of_calls += 1
+        return np.sum(x**2), 2*x
+
+
+@pytest.fixture
+def function_with_gradient():
+    return FunctionWithGradient()
+
+
+def test_memoize_jac_function_before_gradient(function_with_gradient):
+    memoized_function = MemoizeJac(function_with_gradient)
+
+    x0 = np.array([1.0, 2.0])
+    assert_allclose(memoized_function(x0), 5.0)
+    assert_(function_with_gradient.number_of_calls == 1)
```

Comment:
Just use a plain `assert function_with_gradient.number_of_calls == 1` here, not `assert_`.
Same goes for the rest of the tests.

Reply:
Ah, that is good to know. I prefer plain asserts, but according to the [scipy testing guidelines](https://docs.scipy.org/doc/numpy/reference/testing.html#writing-your-own-tests), one is supposed to use `assert_`.

Reply:
https://github.com/scipy/scipy/issues/8583
We really should get around to changing that

Reply:
I changed it to use plain assert.